### PR TITLE
Task/hw 53624 create new destination account

### PR DESCRIPTION
**Pull request by ddhar-hw**

### CODE REVIEW

File: Common/Sources/HyperwalletCoordinatorTypes.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private static var instance: HyperwalletCoordinatorFactory!
+
+    /// Returns the previously initialized instance of the Hyperwallet UI SDK interface object
+    public static var shared: HyperwalletCoordinatorFactory {
+        if instance == nil {
+            self.instance = HyperwalletCoordinatorFactory()
+        }
+        return instance
+    }
+
```
Comment (ymoskovko-epam) [inline Common/Sources/HyperwalletCoordinatorTypes.swift]: If this is a singleton implementation we have to hide the default initializer. 
```
private init() {}
```

File: Common/Sources/HyperwalletCoordinatorTypes.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+/// Hyperwallet'se num for the types of coordinators
```
Comment (ymoskovko-epam) [inline Common/Sources/HyperwalletCoordinatorTypes.swift]: typo

File: Common/Sources/HyperwalletCoordinatorTypes.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            self.instance = HyperwalletCoordinatorFactory()
+        }
+        return instance
+    }
+
+    /// Based on the HyperwalletCoordinatorTypes, determines the coordinator to be returned,
+    /// returns nil if teh module has not been imported in the project
+    ///
+    /// - Parameter hyperwalletCoordinatorType:HyperwalletCoordinatorTypes
+    /// - Returns: The initialized co-ordinator class
```
Comment (ymoskovko-epam) [inline Common/Sources/HyperwalletCoordinatorTypes.swift]: typo

File: Transfer/Sources/CreateTransferController.swift [hunk 4]
```diff
@@ -315,16 +324,23 @@ extension CreateTransferController: CreateTransferView {
...
+        navigateToTransferMethodIfInitialized()
+    }
+
+    private func navigateToTransferMethodIfInitialized() {
+        if let transferMethodCoordinator = selectTransferMethodCoordinator {
+            transferMethodCoordinator.start(initializationData: nil, parentController: self)
+            transferMethodCoordinator.navigate()
+        } else {
+            HyperwalletUtilViews.showAlert(self,
+                                           title: "Error",
```
Comment (ymoskovko-epam) [inline Transfer/Sources/CreateTransferController.swift]: should be ```title: "error".localized(),```

File: Common/Sources/HyperwalletCoordinatorTypes.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    /// Returns the previously initialized instance of the Hyperwallet UI SDK interface object
+    public static var shared: HyperwalletCoordinatorFactory {
+        if instance == nil {
+            self.instance = HyperwalletCoordinatorFactory()
+        }
+        return instance
+    }
+
+    /// Based on the HyperwalletCoordinatorTypes, determines the coordinator to be returned,
+    /// returns nil if teh module has not been imported in the project
```
Comment (jsaini-hw) [inline Common/Sources/HyperwalletCoordinatorTypes.swift]: typo in teh - the

Comment (ddhar-hw) [reply Common/Sources/HyperwalletCoordinatorTypes.swift]: 🤦‍♂

**Pull request merged**